databricks/resource_databricks_ipaccesslist.go

```diff
@@ -0,0 +1,135 @@
+package databricks
+
+// Preview feature: https://docs.databricks.com/security/network/ip-access-list.html
+// REST API: https://docs.databricks.com/dev-tools/api/latest/ip-access-list.html#operation/create-list
+
+import (
+	"log"
+
+	"github.com/databrickslabs/databricks-terraform/client/model"
+	"github.com/databrickslabs/databricks-terraform/client/service"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceIPAccessList() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceIPACLCreate,
+		Read:   resourceIPACLRead,
+		Update: resourceIPACLUpdate,
+		Delete: resourceIPACLDelete,
+
+		Schema: map[string]*schema.Schema{
+			"label": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"list_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						string(model.WhiteList),
+						string(model.BlackList),
+					}, false),
+			},
+			"ip_addresses": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.Any(validation.IsIPv4Address, validation.IsCIDR),
+				},
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceIPACLCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*service.DBApiClient)
+	label := d.Get("label").(string)
+	ipAddresses := d.Get("ip_addresses").([]interface{})
+	listType := d.Get("list_type").(string)
+
+	log.Println("IPACLLists: Calling IP ACL Create")
+	status, err := client.IPAccessLists().Create(
+		convertListInterfaceToString(ipAddresses),
+		label,
+		model.IPAccessListType(listType))
+	log.Printf("IPACLLists: Created as  %v\n", status)
+	if err != nil {
+		log.Printf("IPACLLists:  Creation error %v\n", err)
+		return err
+	}
+
+	log.Println("IPACLLists:  Updating interface")
+	err = updateFromStatus(d, status)
+	log.Printf("IPACLLists:  Updated interface %v\n", d)
+	if err != nil {
+		log.Printf("IPACLLists:  Error updating interface %v\n", err)
+		return err
+	}
```

Comment:
Unless the fields are static and not generated by the "read" api , we should not set anything but the id in the terraform create: example here- https://github.com/terraform-providers/terraform-provider-aws/blob/34482925e20774621ab80fa0b03054514795460e/aws/resource_aws_appmesh_route.go#L293-L313.
Terraform actually uses the set to identify differences. So it should be done after a read action and not a create action. This can identify inconsistencies in what was created and what the hcl schema expects.

Reply:
Interresting, ok.  Removed that section.